<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Run JLOH with a test dataset &mdash; jloh 1.0.0 documentation</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../_static/jquery.js?v=5d32c60e"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../_static/documentation_options.js?v=8d563738"></script>
        <script src="../_static/doctools.js?v=888ff710"></script>
        <script src="../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Modelling SNP density" href="model_snp_density.html" />
    <link rel="prev" title="Quick start" href="quickstart.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: white" >

          
          
          <a href="../index.html" class="icon icon-home">
            jloh
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Get started</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="quickstart.html">Quick start</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Run JLOH with a test dataset</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#get-loh-blocks">Get LOH blocks</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="model_snp_density.html">Modelling SNP density</a></li>
<li class="toctree-l1"><a class="reference internal" href="hybrid_wf.html">Working with hybrids</a></li>
<li class="toctree-l1"><a class="reference internal" href="one_parent.html">Hybrids with only one known parental</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Extraction tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="modules/Extraction/stats.html">jloh stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules/Extraction/g2g.html">jloh g2g</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules/Extraction/extract.html">jloh extract</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Operations tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="modules/Operations/filter.html">jloh filter</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules/Operations/intersect.html">jloh intersect</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules/Operations/chimeric.html">jloh chimeric</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules/Operations/junctions.html">jloh junctions</a></li>
<li class="toctree-l1"><a class="reference internal" href="modules/Operations/cluster.html">jloh cluster</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Visualization tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="modules/Visualization/plot.html">jloh plot</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Simulation tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="modules/Simulation/sim.html">jloh sim</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu"  style="background: white" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">jloh</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Run JLOH with a test dataset</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/usage/run_test_data.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="run-jloh-with-a-test-dataset">
<span id="test-data"></span><h1>Run JLOH with a test dataset<a class="headerlink" href="#run-jloh-with-a-test-dataset" title="Link to this heading"></a></h1>
<p>If you wish to perform a test run to see the output files and their structure, we have prepared a test dataset starting from the NCBI genomic read accession SRR12033503 and the sequence of the chromosome XII of <em>S. paradoxus</em>. This dataset can be found in the <code class="docutils literal notranslate"><span class="pre">test_data</span></code> folder, within the installation directory.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>File</p></th>
<th class="head"><p>Format</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>S_para.chrXII.fa</p></td>
<td><p>FASTA</p></td>
<td><p>Reference sequence</p></td>
</tr>
<tr class="row-odd"><td><p>out.fs.bam</p></td>
<td><p>BAM</p></td>
<td><p>File with mapping results</p></td>
</tr>
<tr class="row-even"><td><p>out.ff.vcf</p></td>
<td><p>VCF</p></td>
<td><p>File with called SNPs</p></td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>A key feature of the VCF file used here is that it contains allele frequency (AF) annotation in the FORMAT (8th) field. This is crucial for <a class="reference internal" href="modules/Extraction/extract.html#jloh-extract"><span class="std std-ref">jloh extract</span></a> to work.</p>
</div>
<section id="get-loh-blocks">
<h2>Get LOH blocks<a class="headerlink" href="#get-loh-blocks" title="Link to this heading"></a></h2>
<p>To infer LOH blocks from the test dataset, you first should model the distribution of SNP densities across the genome with <a class="reference internal" href="modules/Extraction/stats.html#jloh-stats"><span class="std std-ref">jloh stats</span></a>.</p>
<p>To do so, use the following command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>jloh<span class="w"> </span>stats<span class="w"> </span>--vcf<span class="w"> </span>out.ff.vcf
</pre></div>
</div>
<p>This command produces the quantiles of heterozygous and homozygous SNP densities. See <a class="reference internal" href="model_snp_density.html#model-snp-density"><span class="std std-ref">Modelling SNP density</span></a> for details. You must choose a threshold SNP density for heterozygous SNPs and one for homozygous SNPs.</p>
<p>Then, we proceed inferring LOH blocks from the dataset. This is done with the <a class="reference internal" href="modules/Extraction/extract.html#jloh-extract"><span class="std std-ref">jloh extract</span></a> module. We used this command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>jloh<span class="w"> </span>extract<span class="w"> </span><span class="se">\</span>
--vcf<span class="w"> </span>out.ff.vcf<span class="w"> </span>--bam<span class="w"> </span>out.fs.bam<span class="w"> </span>--ref<span class="w"> </span>S_para.chrXII.fa<span class="w"> </span><span class="se">\</span>
--min-snps-kbp<span class="w"> </span><span class="m">2</span>,5<span class="w"> </span>--output-dir<span class="w"> </span>jloh_out
</pre></div>
</div>
<p>The output will be placed in the <code class="docutils literal notranslate"><span class="pre">jloh_out</span></code> directory. The most important file produced by this command is the <code class="docutils literal notranslate"><span class="pre">jloh.LOH_blocks.tsv</span></code> file. This file contains the detected LOH blocks based on the distribution of heterozygous and homozygous SNPs. It is accompanied by a BED file representing the same intervals in a 0-based half-open fashion.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">jloh.LOH_blocks.tsv</span></code> file can be passed to the <a class="reference internal" href="modules/Visualization/plot.html#jloh-plot"><span class="std std-ref">jloh plot</span></a> module to represent LOH graphically with this command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>jloh<span class="w"> </span>plot<span class="w"> </span><span class="se">\</span>
--one-ref<span class="w"> </span><span class="se">\</span>
--loh<span class="w"> </span>jloh_out/jloh.LOH_blocks.tsv<span class="w"> </span><span class="se">\</span>
--het<span class="w"> </span>jloh_out/jloh.exp.het_blocks.bed<span class="w"> </span><span class="se">\</span>
--contrast<span class="w"> </span>max
</pre></div>
</div>
<p>Due to the low variation in the test dataset, we have increased the <code class="docutils literal notranslate"><span class="pre">--contrast</span></code> to the maximum value to highlight potential LOH regions. The <code class="docutils literal notranslate"><span class="pre">--one-ref</span></code> option indicates that the <a class="reference internal" href="modules/Extraction/extract.html#jloh-extract"><span class="std std-ref">jloh extract</span></a> was run in default mode with only one reference FASTA file.</p>
<p>Here’s how the plot should look like.</p>
<img alt="../_images/plot.max_contrast.png" src="../_images/plot.max_contrast.png" />
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="quickstart.html" class="btn btn-neutral float-left" title="Quick start" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="model_snp_density.html" class="btn btn-neutral float-right" title="Modelling SNP density" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Matteo Schiavinato.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>
    <!-- Theme Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-XXXXXXXXXX"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-XXXXXXXXXX', {
          'anonymize_ip': false,
      });
    </script> 

</body>
</html>