<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>jloh chimeric &mdash; jloh 1.1.0 documentation</title>
      <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../../../_static/jquery.js?v=5d32c60e"></script>
        <script src="../../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../../../_static/documentation_options.js?v=fc837d61"></script>
        <script src="../../../_static/doctools.js?v=888ff710"></script>
        <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="jloh junctions" href="junctions.html" />
    <link rel="prev" title="jloh intersect" href="intersect.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: white" >

          
          
          <a href="../../../index.html" class="icon icon-home">
            jloh
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Get started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../quickstart.html">Quick start</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../run_test_data.html">Run JLOH with a test dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../model_snp_density.html">Modelling SNP density</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../hybrid_wf.html">Working with hybrids</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../cancer_wf.html">Working with cancer data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../one_parent.html">Hybrids with only one known parental</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Extraction tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../Extraction/stats.html">jloh stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Extraction/g2g.html">jloh g2g</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Extraction/extract.html">jloh extract</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Extraction/onco_extract.html">jloh onco_extract</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Operations tools</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="filter.html">jloh filter</a></li>
<li class="toctree-l1"><a class="reference internal" href="intersect.html">jloh intersect</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">jloh chimeric</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#description">Description</a></li>
<li class="toctree-l2"><a class="reference internal" href="#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="#parameters">Parameters</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="junctions.html">jloh junctions</a></li>
<li class="toctree-l1"><a class="reference internal" href="cluster.html">jloh cluster</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Visualization tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../Visualization/plot.html">jloh plot</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Simulation tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../Simulation/sim.html">jloh sim</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu"  style="background: white" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">jloh</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">jloh chimeric</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../../_sources/usage/modules/Operations/chimeric.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="jloh-chimeric">
<span id="id1"></span><h1>jloh chimeric<a class="headerlink" href="#jloh-chimeric" title="Link to this heading"></a></h1>
<section id="description">
<h2>Description<a class="headerlink" href="#description" title="Link to this heading"></a></h2>
<p>This module extracts genes from a genome annotation file which contain traces of LOH towards two different haplotypes (REF and ALT).</p>
<p>Biologically, these are likely the result of an LOH breakpoint placed within an intron of a gene, so that part of the gene has the REF haplotype, and part of the gene has the ALT one.</p>
</section>
<section id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Link to this heading"></a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>jloh<span class="w"> </span>chimeric<span class="w"> </span>--blocks-A<span class="w"> </span>&lt;TSV&gt;<span class="w"> </span>--blocks-B<span class="w"> </span>&lt;TSV&gt;<span class="w"> </span>--het<span class="w"> </span>&lt;BED&gt;<span class="w"> </span>--gff<span class="w"> </span>&lt;GFF&gt;<span class="w"> </span><span class="o">[</span>options<span class="o">]</span>
</pre></div>
</div>
</section>
<section id="parameters">
<h2>Parameters<a class="headerlink" href="#parameters" title="Link to this heading"></a></h2>
<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--blocks-A</span> <span class="pre">&lt;PATH&gt;</span></span></dt>
<dd><p>A TSV file produced by <a class="reference internal" href="../Extraction/extract.html#jloh-extract"><span class="std std-ref">jloh extract</span></a>.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--blocks-B</span> <span class="pre">&lt;PATH&gt;</span></span></dt>
<dd><p>Another TSV file produced by another run of <a class="reference internal" href="../Extraction/extract.html#jloh-extract"><span class="std std-ref">jloh extract</span></a>, profiling a different genotype from <code class="docutils literal notranslate"><span class="pre">--blocks-A</span></code>.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--het</span> <span class="pre">&lt;PATH&gt;</span></span></dt>
<dd><p>Heterozygous regions identified by <a class="reference internal" href="../Extraction/extract.html#jloh-extract"><span class="std std-ref">jloh extract</span></a>.</p>
<p>If the extraction was done with <code class="docutils literal notranslate"><span class="pre">--assign-blocks</span></code> it has produced two files with Heterozygous blocks (one per parental genome). In that case, concatenate them together.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--gff</span> <span class="pre">&lt;PATH&gt;</span></span></dt>
<dd><p>GFF3 file containing gene models from the reference FASTA file used to infer LOH blocks.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--out-prefix</span> <span class="pre">&lt;PATH/STR&gt;</span></span></dt>
<dd><p>Pre-pend this prefix to each output file (path allowed).</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--quiet</span> <span class="pre">&lt;BOOL&gt;</span></span></dt>
<dd><p>Suppress warnings.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--feature</span> <span class="pre">&lt;STR&gt;</span></span></dt>
<dd><p>What feature to look for (see GFF format, 3rd column). Examples are “gene”, “exon”, “CDS”, “mRNA”.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--min-overlap</span> <span class="pre">&lt;FLOAT&gt;</span></span></dt>
<dd><p>Minimum fraction [0.0 - 1.0] of a feature that must overlap an LOH block to create an instance of candidate chimeric feature.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--id-attr</span> <span class="pre">&lt;STR&gt;</span></span></dt>
<dd><p>Attribute to search for to get the names of the features analysed (GFF3 last column, e.g. “ID”).</p>
</dd></dl>

</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="intersect.html" class="btn btn-neutral float-left" title="jloh intersect" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="junctions.html" class="btn btn-neutral float-right" title="jloh junctions" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Matteo Schiavinato.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>
    <!-- Theme Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-XXXXXXXXXX"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-XXXXXXXXXX', {
          'anonymize_ip': false,
      });
    </script> 

</body>
</html>