<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>jloh onco_extract &mdash; jloh 1.1.0 documentation</title>
      <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../../_static/css/theme.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="../../../_static/jquery.js?v=5d32c60e"></script>
        <script src="../../../_static/_sphinx_javascript_frameworks_compat.js?v=2cd50e6c"></script>
        <script src="../../../_static/documentation_options.js?v=fc837d61"></script>
        <script src="../../../_static/doctools.js?v=888ff710"></script>
        <script src="../../../_static/sphinx_highlight.js?v=dc90522c"></script>
    <script src="../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="jloh filter" href="../Operations/filter.html" />
    <link rel="prev" title="jloh extract" href="extract.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: white" >

          
          
          <a href="../../../index.html" class="icon icon-home">
            jloh
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Get started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../quickstart.html">Quick start</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../run_test_data.html">Run JLOH with a test dataset</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../model_snp_density.html">Modelling SNP density</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../hybrid_wf.html">Working with hybrids</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../cancer_wf.html">Working with cancer data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../one_parent.html">Hybrids with only one known parental</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Extraction tools</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="stats.html">jloh stats</a></li>
<li class="toctree-l1"><a class="reference internal" href="g2g.html">jloh g2g</a></li>
<li class="toctree-l1"><a class="reference internal" href="extract.html">jloh extract</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">jloh onco_extract</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#description">Description</a></li>
<li class="toctree-l2"><a class="reference internal" href="#usage">Usage</a></li>
<li class="toctree-l2"><a class="reference internal" href="#parameters">Parameters</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#default-mode">Default mode</a></li>
<li class="toctree-l3"><a class="reference internal" href="#single-sample-mode">Single sample mode</a></li>
<li class="toctree-l3"><a class="reference internal" href="#common-parameters">Common Parameters</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#input-output">Input/Output</a></li>
<li class="toctree-l4"><a class="reference internal" href="#variants">Variants</a></li>
<li class="toctree-l4"><a class="reference internal" href="#blocks">Blocks</a></li>
<li class="toctree-l4"><a class="reference internal" href="#misc">Misc</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Operations tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../Operations/filter.html">jloh filter</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Operations/intersect.html">jloh intersect</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Operations/chimeric.html">jloh chimeric</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Operations/junctions.html">jloh junctions</a></li>
<li class="toctree-l1"><a class="reference internal" href="../Operations/cluster.html">jloh cluster</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Visualization tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../Visualization/plot.html">jloh plot</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Simulation tools</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../Simulation/sim.html">jloh sim</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu"  style="background: white" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">jloh</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">jloh onco_extract</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../../_sources/usage/modules/Extraction/onco_extract.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="jloh-onco-extract">
<span id="onco-extract"></span><h1>jloh onco_extract<a class="headerlink" href="#jloh-onco-extract" title="Link to this heading"></a></h1>
<section id="description">
<h2>Description<a class="headerlink" href="#description" title="Link to this heading"></a></h2>
<p>Extract candidate LOH blocks from single-nucleotide polymorphisms (SNPs) called from reads mapped onto a reference genome using cancer data. Data may be a single set of BAM+VCF or a matched tumor/normal pair (recommended).</p>
</section>
<section id="usage">
<h2>Usage<a class="headerlink" href="#usage" title="Link to this heading"></a></h2>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>jloh<span class="w"> </span>onco_extract<span class="w"> </span>--vcfs<span class="w"> </span>&lt;VCF_control&gt;<span class="w"> </span>&lt;VCF_tumor&gt;<span class="w"> </span>--ref<span class="w"> </span>&lt;FASTA&gt;<span class="w"> </span>--bams<span class="w"> </span>&lt;BAM_control&gt;<span class="w"> </span>&lt;BAM_tumor&gt;<span class="w"> </span><span class="o">[</span>options<span class="o">]</span>
</pre></div>
</div>
<p>Or, if using <code class="docutils literal notranslate"><span class="pre">--single-mode</span></code>:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>jloh<span class="w"> </span>onco_extract<span class="w"> </span>--vcf<span class="w"> </span>&lt;VCF&gt;<span class="w"> </span>--ref<span class="w"> </span>&lt;FASTA&gt;<span class="w"> </span>--bam<span class="w"> </span>&lt;BAM&gt;<span class="w"> </span><span class="o">[</span>options<span class="o">]</span>
</pre></div>
</div>
</section>
<section id="parameters">
<h2>Parameters<a class="headerlink" href="#parameters" title="Link to this heading"></a></h2>
<section id="default-mode">
<h3>Default mode<a class="headerlink" href="#default-mode" title="Link to this heading"></a></h3>
<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--vcfs</span> <span class="pre">[&lt;PATH_1&gt;</span> <span class="pre">&lt;PATH_2&gt;]</span></span></dt>
<dd><p>VCF files (space-separated) from control &amp; tumor, in this order.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--bams</span> <span class="pre">[&lt;PATH_1&gt;</span> <span class="pre">&lt;PATH_2&gt;]</span></span></dt>
<dd><p>BAM files from used to call the –vcfs (space-separated).</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--ref</span> <span class="pre">&lt;PATH&gt;</span></span></dt>
<dd><p>FASTA file where reads were mapped.</p>
</dd></dl>

</section>
<section id="single-sample-mode">
<h3>Single sample mode<a class="headerlink" href="#single-sample-mode" title="Link to this heading"></a></h3>
<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--single-mode</span></span></dt>
<dd><p>Activate block assignment mode.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--vcf</span> <span class="pre">&lt;PATH_1&gt;</span></span></dt>
<dd><p>VCF file containing single-nucleotide polymorphisms (SNPs).</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--bam</span> <span class="pre">&lt;PATH_1&gt;</span></span></dt>
<dd><p>BAM file containing read mapping records.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--refs</span> <span class="pre">&lt;PATH_1&gt;</span></span></dt>
<dd><p>FASTA file where reads were mapped.</p>
</dd></dl>

</section>
<section id="common-parameters">
<h3>Common Parameters<a class="headerlink" href="#common-parameters" title="Link to this heading"></a></h3>
<section id="input-output">
<h4>Input/Output<a class="headerlink" href="#input-output" title="Link to this heading"></a></h4>
<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--sample</span> <span class="pre">&lt;STR&gt;</span></span></dt>
<dd><p>Sample name for output files.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--output-dir</span> <span class="pre">&lt;PATH&gt;</span></span></dt>
<dd><p>Path to an output directory (created if not existing)</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--regions</span> <span class="pre">&lt;PATH&gt;</span></span></dt>
<dd><p>Path to a BED file with regions of interest. The BED file <strong>must</strong> contain 4 columns: chromosome, start position, end position, and annotation. Annotation may be anything (gene name, transcript name, exon name, locus) as long as it is an alphanumeric string.</p>
</dd></dl>

</section>
<section id="variants">
<h4>Variants<a class="headerlink" href="#variants" title="Link to this heading"></a></h4>
<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--max-dist</span> <span class="pre">&lt;INT&gt;</span></span></dt>
<dd><p>Maximum distance allowed between SNPs for them to still be retained within the same homozygous / heterozygous block.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--filter-mode</span> <span class="pre">[&quot;all&quot;|&quot;pass&quot;]</span></span></dt>
<dd><p>Either “all” or “pass”. Whether to select only VCF entries that have the <code class="docutils literal notranslate"><span class="pre">PASS</span></code> annotation or not.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--min-af</span> <span class="pre">&lt;FLOAT&gt;</span></span></dt>
<dd><p>Minimum allele frequency to consider a SNP heterozygous. Useful when working with polyploid species.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--max-af</span> <span class="pre">&lt;FLOAT&gt;</span></span></dt>
<dd><p>Maximum allele frequency to consider a SNP heterozygous. Useful when working with polyploid species.</p>
</dd></dl>

</section>
<section id="blocks">
<h4>Blocks<a class="headerlink" href="#blocks" title="Link to this heading"></a></h4>
<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--min-length</span> <span class="pre">&lt;INT&gt;</span></span></dt>
<dd><p>Minimum length of accepted candidate LOH blocks.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--min-snps</span> <span class="pre">&lt;INT&gt;</span></span></dt>
<dd><p>Minimum number of homozygous SNPs to consider a block in the final results. Homozygous SNPs are an indicator of LOH when a paired normal sample is present.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--min-snps-het</span> <span class="pre">&lt;INT&gt;</span></span></dt>
<dd><p>Minimum number of heterozygous SNPs to discard a block in the final results.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--min-frac-cov</span> <span class="pre">&lt;FLOAT&gt;</span></span></dt>
<dd><p>Minimum fraction of positions of a candidate LOH block to include it in the final list.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--hemi</span> <span class="pre">&lt;FLOAT&gt;</span></span></dt>
<dd><p>Threshold of coverage ratio between candidate block and surrounding up/downstream regions, below which a block is considered hemizygous (i.e. carrying only one copy).</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--overhang</span> <span class="pre">&lt;INT&gt;</span></span></dt>
<dd><p>Size of the up/downstream region checked to define zygosity (see <code class="docutils literal notranslate"><span class="pre">--hemi</span></code>).</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--min-overhang</span> <span class="pre">&lt;FLOAT&gt;</span></span></dt>
<dd><p>Fraction of the <code class="docutils literal notranslate"><span class="pre">--overhang</span></code> that must be present to infer zygosity (e.g. at the beginning of a chromosome).</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--merge-uncov</span> <span class="pre">&lt;INT&gt;</span></span></dt>
<dd><p>Number of uncovered positions (bp) separating two blocks that are ignored, producing a merged block.</p>
</dd></dl>

</section>
<section id="misc">
<h4>Misc<a class="headerlink" href="#misc" title="Link to this heading"></a></h4>
<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--sample</span> <span class="pre">&lt;STR&gt;</span></span></dt>
<dd><p>Sample name to include in output files.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--output-dir</span> <span class="pre">&lt;PATH&gt;</span></span></dt>
<dd><p>Path to the output directory.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--threads</span> <span class="pre">&lt;INT&gt;</span></span></dt>
<dd><p>Number of parallel operations performed.</p>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py">
<span class="sig-name descname"><span class="pre">--regions</span> <span class="pre">&lt;PATH&gt;</span></span></dt>
<dd><p>BED file containing regions where blocks shall be searched in. This BED file may be created via <a class="reference internal" href="g2g.html#jloh-g2g"><span class="std std-ref">jloh g2g</span></a> or it may be a custom BED file with regions of interest.</p>
</dd></dl>

</section>
</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="extract.html" class="btn btn-neutral float-left" title="jloh extract" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../Operations/filter.html" class="btn btn-neutral float-right" title="jloh filter" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2024, Matteo Schiavinato.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>
    <!-- Theme Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=G-XXXXXXXXXX"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'G-XXXXXXXXXX', {
          'anonymize_ip': false,
      });
    </script> 

</body>
</html>